File: src/main/java/com/partymakers/shareparty/domain/party/usecase/impl/AddPartyExpenseImpl.java
package com.partymakers.shareparty.domain.party.usecase.impl;

import com.partymakers.shareparty.domain.party.entity.Expense;
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository;
import com.partymakers.shareparty.domain.party.usecase.AddPartyExpense;
import com.partymakers.shareparty.domain.party.usecase.exception.NotFoundException;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class AddPartyExpenseImpl implements AddPartyExpense {

    private final PartyRoomRepository repository;

    @Override
    public void addPartyExpense(Expense expense, long partyId) {
        repository.findById(partyId).ifPresentOrElse( partyRoom -> {
            partyRoom.getExpenses().add(expense); repository.save(partyRoom); },
            () ->{throw new NotFoundException("Party room with id: " + partyId + "not found");} );
    }
}

File: src/main/kotlin/com/partymakers/shareparty/domain/party/usecase/impl/AddPartyExpenseImpl.kt
package com.partymakers.shareparty.domain.party.usecase.impl

import com.partymakers.shareparty.domain.party.entity.Expense
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository
import com.partymakers.shareparty.domain.party.usecase.AddPartyExpense
import com.partymakers.shareparty.domain.party.usecase.exception.NotFoundException

class AddPartyExpenseImpl(
    private val repository: PartyRoomRepository
) : AddPartyExpense {

    override fun addPartyExpense(expense: Expense, partyId: Long) {
        val partyRoom = repository.findById(partyId)
            .orElseThrow { NotFoundException("Party room with id: $partyId not found") }

        val updatedPartyRoom = partyRoom.copy(expenses = partyRoom.expenses + expense)
        repository.save(updatedPartyRoom)
    }
} 